analyzer-core/src/main/java/ffb/analyzer/core/pubsub/Publisher.java

```diff
@@ -0,0 +1,79 @@
+package ffb.analyzer.core.pubsub;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class Publisher {
+    private Object mutex = new Object();
```

Comment:
What does `mutex` mean?

Reply:
It stands for Mutual Exclusion. i was going to call it mutualExclusion, but it looks like mutex is a standard term. If you have a preference, we could also rename it "lock" since that's the purpose it serves.

Reply:
No it's fine. Just haven't heard it before.